lib/swaps/SwapClientManager.ts

```diff
@@ -0,0 +1,224 @@
+   * and waits for the swap clients to initialize.
+   * @returns A promise that resolves upon successful initialization, rejects otherwise.
+   */
+  public async init(): Promise<void> {
+    const initPromises = [];
+    // setup LND clients and initialize
+    for (const currency in this.config.lnd) {
+      const lndConfig = this.config.lnd[currency]!;
+      if (!lndConfig.disable) {
+        const lndClient = new LndClient(
+          lndConfig,
+          currency,
+          (SwapClientManager.wrapLndLogger(this.loggers.lnd, currency) as Logger),
+        );
+        this.swapClients.set(currency, lndClient);
+        initPromises.push(lndClient.init());
+      }
+    }
+    // setup raiden client and connect if configured
+    const raidenEnabled = !this.config.raiden.disable;
+    if (raidenEnabled) {
+      this.raidenClient = new RaidenClient(this.config.raiden, this.loggers.raiden);
+      initPromises.push(this.raidenClient.init());
+    }
+    // bind event listeners
+    this.bind();
+    await Promise.all(initPromises);
+    // associate swap clients with currencies managed by raiden client
+    if (raidenEnabled) {
+      for (const currency of this.raidenClient.tokenAddresses.keys()) {
+        this.swapClients.set(currency, this.raidenClient);
+      }
+    }
+  }
+
+  /**
+   * Gets a swap client instance.
+   * @param currency a currency that the swap client is linked to.
+   * @returns swap client instance upon success, throws otherwise.
+   */
+  public get(currency: string): BaseClient {
+    const swapClient = this.swapClients.get(currency);
+    if (!swapClient) {
+      throw errors.SWAP_CLIENT_NOT_FOUND(currency);
+    }
+    return swapClient;
+  }
+
+  /**
+   * Adds a new swap client and currency association.
+   * @param currency a currency that should be linked with a swap client.
+   * @returns Nothing upon success, throws otherwise.
+   */
+  public add(currency: Currency): void {
+    if (currency.swapClient === SwapClient.Raiden && currency.tokenAddress) {
+      this.swapClients.set(currency.id, this.raidenClient);
+      this.raidenClient.tokenAddresses.set(currency.id, currency.tokenAddress);
+    } else if (currency.swapClient === SwapClient.Lnd) {
+      // in case of lnd we check if the configuration includes swap client
+      // for the specified currency
+      let hasCurrency = false;
+      for (const lndCurrency in this.config.lnd) {
+        if (lndCurrency === currency.id) {
+          hasCurrency = true;
+        }
+      }
+      // adding a new lnd client at runtime is currently not supported
+      if (!hasCurrency) {
+        throw errors.SWAP_CLIENT_NOT_CONFIGURED(currency.id);
+      }
+    }
+  }
+
+  /**
+   * Removes a new swap client and currency association.
+   * @param currency a currency that should be unlinked from a swap client.
+   * @returns Nothing upon success, throws otherwise.
+   */
+  public remove(currencyId: string): void {
```

Comment:
To match the @param comment above (and how we're usually calling this throughout the codebase) I'd rename the param to `currency`.